libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -31,13 +51,10 @@ import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.flow.flowOf
 import kotlinx.coroutines.launch
 import java.lang.IllegalStateException
+import java.util.concurrent.CopyOnWriteArraySet
 
 @ExperimentalCoroutinesApi
-class NavigationView : ConstraintLayout, LifecycleObserver {
-
-    val mapView: MapView
-    lateinit var navigationViewModel: NavigationViewModel
-        private set
+class NavigationView : ConstraintLayout {
 
     private val activity: FragmentActivity by lazy {
         try {
```

Comment:
What is the reason for providing `NavigationView` rather than `NavigationFragment`? It doesn't seem right to have all this logic in a `View`. Also using activity as a `LifecycleOwner` looks really error-prone and most likely won't let the user use this `NavigationView` in a `Fragment`. Imagine the `Fragment` with a `NavigationView` gets destroyed while the activity is still running – `MapboxNavigation` will continue to work and thus we have a leak.

Reply:
agree, we already [had this issue](https://github.com/mapbox/mapbox-navigation-android/pull/3012) at `v1.x`. LifecycleOwner has to be provided from outside

Reply:
The fact that we are using FragmentActivity rather than AppCompatActivity will allow us to use it in a fragment. But it's worth trying to get lifecycle owner from outside.

Reply:
But why don't we provide a `NavigationFragment` instead? It has its own lifecycle out of the box. 
Fragments are fully independent components that can be used in many different ways and we still won't have to make any assumptions or rely on the user the provide the correct lifecycle.

Reply:
I don't think implementing this as a fragment is appropriate. A lot of developers don't like to use fragments. Forcing them to use a fragment is too opinionated. By implementing this in a view the developers can include it in any part of their layout.

Reply:
Since `NavigationView` is requiring and activity (FragmentActivity) the lifecycle owner is that activity.  The view model that owns the reference to `MapboxNavigation` is tied to that same activity.  The `MapboxNavigation` reference will live as long as the view model associated with the activity so the Android OS will manage its lifecycle. There are no references to `MapboxNavigation` outside of that view model and `MapboxNavigation` doesn't reference anything outside of the view model that hosts it.

Reply:
Basically fragments are the only way to correctly provide a combination of UI and logic with one self-sufficient component. 

> I don't think implementing this as a fragment is appropriate. A lot of developers don't like to use fragments. Forcing them to use a fragment is too opinionated. By implementing this in a view the developers can include it in any part of their layout. 

Well, fragments can be included in any part of the layout, too. Having `NavigationFragment` doesn't force you to use fragments anywhere else. I'd actually say that current implementation is also opinionated, because it forces the developer to have a separate activity for the `NavigationView` if they want to show it later, e.g. after the user is logged in, or hide it, when the user is logged out. 

> Since `NavigationView` is requiring and activity (FragmentActivity) the lifecycle owner is that activity. The view model that owns the reference to `MapboxNavigation` is tied to that same activity. The `MapboxNavigation` reference will live as long as the view model associated with the activity so the Android OS will manage its lifecycle. There are no references to `MapboxNavigation` outside of that view model and `MapboxNavigation` doesn't reference anything outside of the view model that hosts it.

That forces the `NavigationView` to live as long as activity lives. While it is possible to add and remove the `NavigationFragment` from the activity at any time and its lifecycle will be aware of that.

Reply:
The `NavigationView` class has already been merged into main. The scope of this PR is for issue #1258. There was refactoring done for managing `MapboxNavigation` as well.  

Whether or not the drop-in UI should be implemented as a fragment or not is certainly a discussion the team can have but it's outside the scope of this PR.